packages/email-service/src/app/app.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Body, Controller, Get, Post, Res } from '@nestjs/common';
+import { Response } from 'express';
+import { AppService } from './app.service';
+
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  @Get()
+  getData() {
+    return this.appService.getData();
+  }
+
+  @Post('/send-overdue-notify')
+  async sendOverdueNotify(@Body() receivers: string[], @Res() res: Response) {
+    if (!receivers.length) return res.status(404).send();
+    const sentOk = await this.appService.sendOverdueNotify(receivers);
```

Comment:
Generally, would approve your approach of having a separate service. But let's consider following changes - there is should be a some queue between BE app and a mail service. So BE could post emails to the queue and instantly receive 200 OK. Email service pulls emails from that queue and sends them in order. The queue could be part of email-service. (e.g. https://github.com/bee-queue/bee-queue)